### PR TITLE
started working on the layout

### PR DESCRIPTION
**Pull request by tomasmilkintas**
Started to work on the layout. Refactored stories somewhat. Slowly visualising how things will look from individual pieces. More logic and less hard-coded stuff to follow

### CODE REVIEW

File: src/API/Authentication.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import fire from "./config/Fire";
```
Comment (RVerma90) [inline src/API/Authentication.js:1]: fire -> firebase :)

File: src/API/Authentication.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+export const loginHandler = (email, password) => {
```
Comment (RVerma90) [inline src/API/Authentication.js:3]: Nice!

File: src/API/Authentication.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        .createUserWithEmailAndPassword(email, password)
+        .then((user) =>
+            fire
+                .auth()
+                .currentUser.getIdToken(true)
+                .then((idToken) => console.log(idToken))
+                .catch((err) => console.log(err))
+        )
+        .catch((err) => console.log(err.message));
+    console.log(email, password);
```
Comment (RVerma90) [inline src/API/Authentication.js:21]: Let's remove consoles :)

File: src/API/Tasks.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import React, { Component } from "react";
+
+import List from "../stories/List/index";
+import ListItem from "../stories/List/ListItem/index";
+import Form from "../stories/Form/index";
+import Button from "../stories/Button";
+import Input from "../stories/Input";
+
+class Tasks extends Component {
+    taskAddHandler = () => {};
```
Comment (RVerma90) [inline src/API/Tasks.js:10]: Awesome Scaffolding setup!

File: src/API/config/Fire.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    authDomain: process.env.REACT_APP_AUTH_DOMAIN,
+    databaseURL: process.env.REACT_APP_DATABASE_URL,
+    projectId: process.env.REACT_APP_PROJECT_ID,
+    storageBucket: process.env.REACT_APP_STORAGE_BUCKET,
+    messagingSenderId: process.env.REACT_APP_MESSAGING_SENDER_ID,
+    appId: process.env.REACT_APP_APP_ID,
+    measurementId: process.env.REACT_APP_MEASUREMENT_ID,
+};
+
+const fire = firebase.initializeApp(firebaseConfig);
```
Comment (RVerma90) [inline src/API/config/Fire.js:14]: firebaseInit

File: src/API/Authentication.js (hunk 1 above)
Comment (RVerma90) [reply src/API/Authentication.js:1]: *firebaseInit

File: src/containers/Authentication/Login/index.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    }
+
+    render() {
+        const { redirect } = this.state;
+
+        if (redirect) {
+            return <Redirect to="/signup" />;
+        }
+        return (
+            <div>
```
Comment (RVerma90) [inline src/containers/Authentication/Login/index.js:43]: No need for the outer div

File: src/containers/Authentication/Register/Index.js [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+    }
+
+    render() {
+        const { redirect } = this.state;
+
+        if (redirect) {
+            return <Redirect to="/login" />;
+        }
+        return (
+            <div>
```
Comment (RVerma90) [inline src/containers/Authentication/Register/Index.js:43]: No need to outer div

File: src/containers/Authentication/Register/Index.js [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+                        onChange={(e) => this.changeHandler(e)}
+                        placeholder="Your Password"
+                        type="password"
+                        name="password"
+                        value={this.state.password}
+                    />
+                    <Button onClick={(e) => this.signup(e)}>Sign up</Button>
+                    <p>
+                        Got an account?{" "}
+                        <Link onClick={() => this.clicked()} to="/login">
```
Comment (RVerma90) [inline src/containers/Authentication/Register/Index.js:63]: clicked -> handleOnClick

File: src/containers/Authentication/Register/Index.js [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+    signup(event) {
+        event.preventDefault();
+        signupHandler(this.state.email, this.state.password);
+    }
+
+    changeHandler(event) {
+        this.setState({ [event.target.name]: event.target.value });
+    }
+
+    clicked() {
```
Comment (RVerma90) [inline src/containers/Authentication/Register/Index.js:32]: clicked -> handleOnClick

File: src/containers/Tasks/Home/index.js [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+    render() {
+        const { redirect } = this.state;
+
+        if (redirect) {
+            return <Redirect to="/" />;
+        }
+        return (
+            // visualising the home
+            <div>
+                <div id="status-bars-top"></div>
```
Comment (RVerma90) [inline src/containers/Tasks/Home/index.js:36]: Consider creating a Storybook component

File: src/containers/Tasks/Home/index.js [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+
+                <Header id="header"></Header>
+
+                <div id="logo"></div>
+
+                {/* if no tasks - */}
+                <p id="text">
```
Comment (RVerma90) [inline src/containers/Tasks/Home/index.js:43]: Create Generic Storybook component !

File: src/containers/Welcome/index.js [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+import { BrowserRouter, Route, Switch, Link } from "react-router-dom";
+
+import Register from "../Authentication/Register/Index";
+import Login from "../Authentication/Login";
+
+class Welcome extends Component {
+    render() {
+        return (
+            <BrowserRouter>
+                <h1>Welcome!</h1>
```
Comment (RVerma90) [inline src/containers/Welcome/index.js:11]: Storybook H1 as a main TitleText

File: src/stories/Form/component.js [hunk 13]
```diff
@@ -1,9 +1,13 @@
 import styled from "styled-components";
 
 const StyledForm = styled.form`
-    width: 100px;
-    height: 100px;
-    border: 2px solid teal;
+    width: 400px;
+    height: 400px;
+    border: 4px solid teal;
```
Comment (RVerma90) [inline src/stories/Form/component.js:6]: Teal is worth adding into theme.colours

File: src/stories/Theme/index.js [hunk 14]
```diff
@@ -3,7 +3,7 @@ import React from "react";
 import StyledTheme from "./component";
 
 const Theme = (props) => {
-    return <StyledTheme {...props}>{props.children}</StyledTheme>;
+    return <StyledTheme {...props} />;
```
Comment (RVerma90) [inline src/stories/Theme/index.js:6]: Nice :)

File: src/hoc/Layout/Layout.js [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+import Aux from "../Aux/Aux";
+
+import fire from "../../API/config/Fire";
+
+import Welcome from "../../containers/Welcome";
+// import Login from "../../containers/Authentication/Login";
+
+import Home from "../../containers/Tasks/Home/index";
+
+class Layout extends Component {
```
Comment (RVerma90) [inline src/hoc/Layout/Layout.js:12]: The whole component logic and file and be removed and replaced in app.js

As a fellow developer when I go to your app.js it's helpful if I can see the auth logic to conditionally render  the app. 
This is a little over engineered

File: src/hoc/Aux/Aux.js [hunk 16]
```diff
@@ -0,0 +1,3 @@
+const aux = (props) => props.children;
+
+export default aux;
```
Comment (RVerma90) [inline src/hoc/Aux/Aux.js:3]: Not very clear of the purpose of this component.

Consider removing all together and handle auth state logic in app.js as per comment for layout/layout.js

File: src/App.js [hunk 17]
```diff
@@ -1,20 +1,34 @@
...
-        </div>
-    );
+import Layout from "./hoc/Layout/Layout";
+import Login from "./containers/Authentication/Login";
+import Register from "./containers/Authentication/Register/Index";
+import Home from "./containers/Tasks/Home";
+
+class App extends Component {
+    render() {
+        let routes = (
```
Comment (RVerma90) [inline src/App.js:13]: Instead of creating this variable you can put it right inside where the div currently sits.

File: src/App.js [hunk 18]
```diff
@@ -1,20 +1,34 @@
...
+                    Register
+                </Route>
+                <Route path="/home" component={Home} />
+
+                <Redirect to="/" />
+            </Switch>
+        );
+        return (
+            <div className="App">
+                <Layout>{routes}</Layout>
```
Comment (RVerma90) [inline src/App.js:28]: Handle Auth conditional logic inside this component, it is not very clear that Layout actually does this.

File: src/containers/Authentication/Login/index.js (hunk 6 above)
Comment (tomasmilkintas) [reply src/containers/Authentication/Login/index.js:43]: done

File: src/containers/Authentication/Register/Index.js (hunk 9 above)
Comment (tomasmilkintas) [reply src/containers/Authentication/Register/Index.js:32]: replaced

File: src/API/config/Fire.js (hunk 5 above)
Comment (tomasmilkintas) [reply src/API/config/Fire.js:14]: replaced

File: src/API/Authentication.js (hunk 1 above)
Comment (tomasmilkintas) [reply src/API/Authentication.js:1]: replaced

File: src/stories/Form/component.js (hunk 13 above)
Comment (tomasmilkintas) [reply src/stories/Form/component.js:6]: Agreed, done

File: src/hoc/Layout/Layout.js (hunk 15 above)
Comment (tomasmilkintas) [reply src/hoc/Layout/Layout.js:12]: agreed, replaced to app.js

File: src/hoc/Aux/Aux.js (hunk 16 above)
Comment (tomasmilkintas) [reply src/hoc/Aux/Aux.js:3]: done

File: src/App.js (hunk 18 above)
Comment (tomasmilkintas) [reply src/App.js:28]: done

File: src/App.js (hunk 17 above)
Comment (tomasmilkintas) [reply src/App.js:13]: done

File: src/API/Authentication.js (hunk 3 above)
Comment (tomasmilkintas) [reply src/API/Authentication.js:21]: removed

File: src/containers/Welcome/index.js (hunk 12 above)
Comment (tomasmilkintas) [reply src/containers/Welcome/index.js:11]: done

File: src/containers/Tasks/Home/index.js (hunk 11 above)
Comment (tomasmilkintas) [reply src/containers/Tasks/Home/index.js:43]: done

File: src/containers/Tasks/Home/index.js (hunk 10 above)
Comment (tomasmilkintas) [reply src/containers/Tasks/Home/index.js:36]: done for both top and bottom

File: src/App.js [hunk 19]
```diff
@@ -1,20 +1,57 @@
...
+import Welcome from "./containers/Welcome";
+import PasswordRecovery from "./containers/Authentication/ForgotPassword";
+
+class App extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            user: {},
+            name: "",
```
Comment (RVerma90) [inline src/App.js:19]: firstName lastName

File: src/containers/Authentication/Register/Index.js [hunk 20]
```diff
@@ -0,0 +1,83 @@
...
+import TitleText from "../../../stories/TitleText";
+
+class Register extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            email: "",
+            password: "",
+            name: "",
```
Comment (RVerma90) [inline src/containers/Authentication/Register/Index.js:19]: firstName lastName

File: src/containers/Tasks/Home/index.js [hunk 21]
```diff
@@ -0,0 +1,72 @@
...
+
+        if (redirect) {
+            return <Redirect to="/" />;
+        }
+        return (
+            // visualising the home
+            <Container>
+                <StatusBarTop>Icons</StatusBarTop>
+
+                <TitleText id="name">Hello, {this.state.name}</TitleText>
```
Comment (RVerma90) [inline src/containers/Tasks/Home/index.js:56]: {this.state.firstName}

**Pull request merged**